modules/pam_nologin/pam_nologin.c

```diff
@@ -99,21 +99,25 @@ static int perform_check(pam_handle_t *pamh, struct opt_s *opts)
 	    goto clean_up_fd;
 	}
 
-	mtmp = malloc(st.st_size+1);
-	if (!mtmp) {
-	    pam_syslog(pamh, LOG_CRIT, "out of memory");
-	    retval = PAM_BUF_ERR;
-	    goto clean_up_fd;
-	}
-
-	if (pam_modutil_read(fd, mtmp, st.st_size) == st.st_size) {
+	/* Don't print anything if the message is empty, will only
+	   disturb the output with empty lines */
+	if (st.st_size > 0) {
+	    mtmp = malloc(st.st_size+1);
+	    if (!mtmp) {
+	        pam_syslog(pamh, LOG_CRIT, "out of memory");
+		retval = PAM_BUF_ERR;
+		goto clean_up_fd;
+	    }
+
+	    if (pam_modutil_read(fd, mtmp, st.st_size) == st.st_size) {
 		mtmp[st.st_size] = '\0';
 		(void) pam_prompt (pamh, msg_style, NULL, "%s", mtmp);
```

Comment:
The content inside the block appears to have the same level of indentation as that outside the block.

Reply:
Don't trust your webbrowser...